src/data/settings-script.js

```diff
@@ -1,80 +1,229 @@
+    const changeWifiLink = document.getElementById("change-wifi-link");
+    const mqttServerField = document.getElementById("mqtt-server");
+    const mqttPortField = document.getElementById("mqtt-port");
+    const mqttUserField = document.getElementById("mqtt-user");
+    const mqttPasswordField = document.getElementById("mqtt-password");
+
+    function toggleSettings() {
+        const modal = document.getElementById("settingsModal");
+        modal.classList.toggle("hidden");
+
+        // Prevent background scroll when modal is open
+        if (!modal.classList.contains("hidden")) {
+            document.body.classList.add("modal-open");
+        } else {
+            document.body.classList.remove("modal-open");
+        }
+    }
+
+    function enableWifiEdit() {
+        ssidField.readOnly = false;
+        document.getElementById("wifi-password-group").style.display = "block";
+        changeWifiLink.style.display = "none";
+    }
+
+    function loadConfig() {
+        const timezoneOptions = [
+            { label: "UTC (GMT±0)", value: "UTC0" },
+            { label: "Central European Time (GMT+1)", value: "CET-1CEST,M3.5.0/2,M10.5.0/3" },
+            { label: "Eastern European Time (GMT+2)", value: "EET-2EEST,M3.5.0/3,M10.5.0/4" },
+            { label: "Moscow / East Africa (GMT+3)", value: "MSK-3" },
+            { label: "Gulf Standard Time (GMT+4)", value: "GST-4" },
+            { label: "Pakistan Standard Time (GMT+5)", value: "PKT-5" },
+            { label: "Bangladesh / Bhutan (GMT+6)", value: "BST-6" },
+            { label: "Thailand / Vietnam (GMT+7)", value: "ICT-7" },
+            { label: "China / Singapore / Perth (GMT+8)", value: "CST-8" },
+            { label: "Japan / Korea (GMT+9)", value: "JST-9" },
+            { label: "Sydney / Guam (GMT+10)", value: "AEST-10AEDT,M10.1.0,M4.1.0/3" },
+            { label: "Solomon Islands (GMT+11)", value: "SBT-11" },
+            { label: "New Zealand / Fiji (GMT+12)", value: "NZST-12NZDT,M9.5.0,M4.1.0/3" },
+            { label: "Azores (GMT-1)", value: "AZOT1AZOST,M3.5.0/0,M10.5.0/1" },
+            { label: "South Georgia (GMT-2)", value: "GST2" },
+            { label: "Brazil / Argentina (GMT-3)", value: "BRT3BRST,M10.3.0/0,M2.3.0/0" },
+            { label: "Atlantic / Venezuela (GMT-4)", value: "AST4" },
+            { label: "Eastern US / Colombia (GMT-5)", value: "EST5EDT,M3.2.0/2,M11.1.0/2" },
+            { label: "Central US / Mexico City (GMT-6)", value: "CST6CDT,M3.2.0/2,M11.1.0/2" },
+            { label: "Mountain US / Canada (GMT-7)", value: "MST7MDT,M3.2.0/2,M11.1.0/2" },
+            { label: "Pacific US / Canada (GMT-8)", value: "PST8PDT,M3.2.0/2,M11.1.0/2" },
+            { label: "Alaska (GMT-9)", value: "AKST9AKDT,M3.2.0/2,M11.1.0/2" },
+            { label: "Hawaii / Tahiti (GMT-10)", value: "HST10" },
+            { label: "Niue (GMT-11)", value: "NUT11" },
+            { label: "Baker Island (GMT-12)", value: "BIT12" },
+        ];
```

Comment:
I like that you ended up with a minimal list of timezones. I went with the full list, which is definitely overkill and wastes a lot of space: https://github.com/jhoff/Split-Flap-Display/blob/main/src/web/timezones.json